services/src/main/java/org/keycloak/services/x509/AbstractClientCertificateFromHttpHeadersLookup.java

```diff
@@ -114,27 +115,35 @@ protected X509Certificate getCertificateFromHttpHeader(HttpRequest request, Stri
 
 
     @Override
-    public X509Certificate[] getCertificateChain(HttpRequest httpRequest) throws GeneralSecurityException {
+    public final X509Certificate[] getCertificateChain(HttpRequest httpRequest) throws GeneralSecurityException {
+        if (!httpRequest.isProxyTrusted()) {
+            logger.warnf("HTTP header \"%s\" is not trusted", httpRequest);
```

Comment:
I wonder if it might be security risk to print the whole request as it contains the headers, incl. the cert?